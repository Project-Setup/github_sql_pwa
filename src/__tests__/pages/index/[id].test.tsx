import React from 'react';
import { ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Page, { getStaticProps } from 'pages/index/[id]';
import mountWithStore from '__tests__/common/mountWithStore';

describe('Page', () => {
  const testProp = {
    test: 'test',
  };

  let TestPageWrapper: ReactWrapper;

  beforeAll(async () => {
    await act(async () => {
      /* eslint-disable-next-line react/jsx-props-no-spreading */
      TestPageWrapper = mountWithStore(<Page id="testId" {...testProp} />)();
    });
    // TestPageWrapper.update();
  });

  it('should render without throwing an error', () => {
    expect(TestPageWrapper).toMatchSnapshot();
  });

  it('should render query param correctly', () => {
    expect(TestPageWrapper.find('p').at(0).text()).toContain('testId');
  });

  it('should render appProps correctly', () => {
    expect(TestPageWrapper.find('p').at(1).text()).toContain(
      JSON.stringify(testProp)
    );
  });

  it('should render page process env correctly', () => {
    expect(TestPageWrapper.find('p').at(2).text()).toContain(
      process.env.NEXT_PUBLIC_TEST_PAGE_VAR
    );
  });
});

describe('getStaticProps', () => {
  const testParams = { id: 'testParams' };
  let testResult: any;
  beforeAll(async () => {
    testResult = await getStaticProps({ params: testParams });
  });

  it('should retrieve correct params', async () => {
    expect(testResult).toHaveProperty(['props', 'id'], 'testParams');
  });

  it('should retrieve correct process env', async () => {
    expect(testResult).toHaveProperty(
      ['props', 'staticPageProcessEnv'],
      process.env.TEST_STATIC_PAGE_PROP
    );
  });
});
