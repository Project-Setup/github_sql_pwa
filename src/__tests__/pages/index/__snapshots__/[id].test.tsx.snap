// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render without throwing an error 1`] = `
<Provider
  store={
    Object {
      "addReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectReducers": [Function],
      "reducer": Object {
        "text": [Function],
      },
      "removeReducers": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      "substituteReducers": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <IndexPage
    id="testId"
    test="test"
  >
    <main>
      <ManifestHead
        hrefCanonical="/index/testId"
        title="github_pwa dynamic-path"
      >
        <Head>
          <Component
            handleStateChange={null}
            inAmpMode={false}
            reduceComponentsToState={[Function]}
          />
        </Head>
      </ManifestHead>
      <p>
        Param: testId
      </p>
      <p>
        Props from _app.tsx: {"test":"test"}
      </p>
      <p>
        Page Process Env: TEST_TEST_PAGE_PROP
      </p>
      <TextDivWrap>
        <DynamicStoreWrap
          callbackOnMount={[Function]}
          callbackOnUnmount={[Function]}
        >
          <DynamicStore
            callbackOnMount={[Function]}
            callbackOnUnmount={[Function]}
            store={
              Object {
                "addReducer": [Function],
                "dispatch": [Function],
                "getState": [Function],
                "injectReducers": [Function],
                "reducer": Object {
                  "text": [Function],
                },
                "removeReducers": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                "substituteReducers": [Function],
                Symbol(observable): [Function],
              }
            }
          />
        </DynamicStoreWrap>
      </TextDivWrap>
      <PrefixedLink
        href="/"
      >
        <Link
          as="/"
          href="/"
        >
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            index
          </a>
        </Link>
      </PrefixedLink>
    </main>
  </IndexPage>
</Provider>
`;
